Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/Uniform_sizing_field.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020 GeometryFactory (France)
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Jane Tournois
+
+#ifndef CGAL_PMP_REMESHING_UNIFORM_SIZING_FIELD_H
+#define CGAL_PMP_REMESHING_UNIFORM_SIZING_FIELD_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/Sizing_field.h>
+
+#include <CGAL/number_utils.h>
+
+namespace CGAL
+{
+namespace Polygon_mesh_processing
+{
+template <class PolygonMesh>
+class Uniform_sizing_field : public CGAL::Sizing_field<PolygonMesh>
+{
+private:
+  typedef CGAL::Sizing_field<PolygonMesh> Base;
+
+public:
+  typedef typename Base::FT         FT;
+  typedef typename Base::Point_3    Point_3;
+  typedef typename Base::halfedge_descriptor halfedge_descriptor;
+  typedef typename Base::vertex_descriptor   vertex_descriptor;
+
+  Uniform_sizing_field(const FT& size, const PolygonMesh& pmesh)
+    : m_sq_short( CGAL::square(4./5. * size))
+    , m_sq_long(  CGAL::square(4./3. * size))
+    , m_pmesh(pmesh)
+  {}
+
+private:
+  FT sqlength(const vertex_descriptor va,
+              const vertex_descriptor vb) const
+  {
+    typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::const_type
+      vpmap = get(CGAL::vertex_point, m_pmesh);
+    return FT(CGAL::squared_distance(get(vpmap, va), get(vpmap, vb)));
+  }
+
+  FT sqlength(const halfedge_descriptor& h) const
+  {
+    return sqlength(target(h, m_pmesh), source(h, m_pmesh));
+  }
+
+public:
+  //todo ip: rewrite to remove this?
+  void calc_sizing_map() const {}
+  void update_sizing_map(const vertex_descriptor& vnew) const {}
+  double get_sizing(vertex_descriptor v) const {return 1.;}
```

Comment:
Not that it will make a big difference but I think this could be made `static constexpr`

Reply:
Last night, I wrote a few constexpr's with std::enable_if in the implementation so that these could be removed altogether. Will test and push it this afternoon.